src/VaultSharp/V1/SecretsEngines/TOTP/ITOTPSecretsEngine.cs

```diff
@@ -46,5 +46,68 @@ public interface ITOTPSecretsEngine
         /// The secret with the <see cref="TOTPCode" /> as the data.
         /// </returns>
         Task<Secret<TOTPCodeValidity>> ValidateCodeAsync(string keyName, string code, string mountPoint = SecretsEngineDefaultPaths.TOTP, string wrapTimeToLive = null);
+
+        /// <summary>
+        /// Creates a new key definition where Vault acts as TOTP provider
+        /// </summary>
+        /// <param name="keyName"><para>[required]</para>
+        /// Specifies the name of the key to create credentials against.
+        /// </param>
+        /// <param name="issuer"><para>[required]</para>
+        /// Specifies the name of the key’s issuing organization.
+        /// </param>
+        /// <param name="accountName"><para>[required]</para>
+        /// Specifies the name of the account associated with the key.
+        /// </param>
+        /// <param name="mountPoint"><para>[optional]</para>
+        /// The mount point for the Nomad backend. Defaults to <see cref="SecretsEngineDefaultPaths.TOTP" />
+        /// Provide a value only if you have customized the Nomad mount point.</param>
+        /// <returns>
+        /// The secret with the <see cref="TOTPProvider" /> as the data.
+        /// </returns>
+        Task<Secret<TOTPProvider>> CreateTOTPProviderKeyAsync(string keyName, string issuer, string accountName, string mountPoint = SecretsEngineDefaultPaths.TOTP);
```

Comment:
can you please wrap these primitive parameters into a request object?

Also, it would be nice to have the request object have common fields, and then an abstract field that concretes into "vault generated key options" and "non vault generated key options.
That way, a caller is only specifying relevant input values.

We can do all the transformation when near the wire for serialization.

Reply:
please add code samples to the readme as well